### ISSUE TITLE
NSGAII errors with combinations of numerical and non-numerical categorical suggestions

### ISSUE THREAD

**Issue by tvdboom** [optuna/optuna#4172]
### Expected behavior

When combining str categorical hyperparameters with numerical, the sampler converts the numerical suggestion to string, which makes the study fail. The issue happens here

https://github.com/optuna/optuna/blob/016795d9b81f860a065688818908d187120e29ad/optuna/samplers/nsgaii/_crossover.py#L40-L46

because a  numpy array can't have mixed types and therefore forces the numerical value to a string.

### Environment

- Optuna version:3.0.3
- Python version:3.10.2
- OS:Windows-10-10.0.19044-SP0
- Sklearn version:1.1.3

### Error messages, stack traces, or logs

```shell
Traceback (most recent call last):
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\IPython\core\interactiveshell.py", line 3378, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-df714957fb70>", line 1, in <module>
    runfile('C:\\Users\\Mavs\\Documents\\Python\\ATOM\\test.py', wdir='C:\\Users\\Mavs\\Documents\\Python\\ATOM')
  File "C:\Program Files\JetBrains\PyCharm 2022.1.3\plugins\python\helpers\pydev\_pydev_bundle\pydev_umd.py", line 198, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm 2022.1.3\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "C:\Users\Mavs\Documents\Python\ATOM\test.py", line 188, in <module>
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\study\study.py", line 419, in optimize
    _optimize(
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\study\_optimize.py", line 66, in _optimize
    _optimize_sequential(
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\study\_optimize.py", line 160, in _optimize_sequential
    frozen_trial = _run_trial(study, func, catch)
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\study\_optimize.py", line 187, in _run_trial
    trial = study.ask()
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\study\study.py", line 514, in ask
    trial = trial_module.Trial(self, trial_id)
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\trial\_trial.py", line 54, in __init__
    self._init_relative_params()
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\trial\_trial.py", line 63, in _init_relative_params
    self.relative_params = self.study.sampler.sample_relative(
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\samplers\nsgaii\_sampler.py", line 211, in sample_relative
    child_params = perform_crossover(
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\samplers\nsgaii\_crossover.py", line 115, in perform_crossover
    if _is_contained(child_params, search_space):
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\samplers\nsgaii\_crossover.py", line 160, in _is_contained
    if not param_distribution._contains(param_distribution.to_internal_repr(param)):
  File "C:\Users\Mavs\Documents\Python\ATOM\venv310\lib\site-packages\optuna\distributions.py", line 516, in to_internal_repr
    raise ValueError(f"'{param_value_in_external_repr}' not in {self.choices}.")
ValueError: '2' not in (None, 2, 3).
```

### Steps to reproduce

```python
import optuna
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split
from sklearn.datasets import load_breast_cancer

X, y = load_breast_cancer(return_X_y=True)

X_train, X_test, y_train, y_test = train_test_split(X, y)

def objective(trial):
    criterion = trial.suggest_categorical("criterion", ["gini", "entropy"])
    max_depth = trial.suggest_categorical("max_depth", [None, 2, 3])

    return DecisionTreeClassifier(criterion=criterion, max_depth=max_depth).fit(X_train, y_train).score(X_test, y_test)


study = optuna.create_study(sampler=optuna.samplers.NSGAIISampler())
study.optimize(objective, n_trials=100)
```

**Comment by c-bata** [comment]
@tvdboom Thank you for reporting the bug. This bug seems to be related with #3982 and probably  fixed at #4035. Could you try the `master` branch of Optuna like the following?

```
$  pip install git+https://github.com/optuna/optuna.git
```

**Comment by tvdboom** [comment]
yes, that fixed it. Thanks for the quick reply. closing.
